src/Motor.Extensions.Hosting.RabbitMQ/RabbitMQConnectionFactory.cs

```diff
@@ -1,83 +1,57 @@
 using System;
-using System.Linq;
+using System.ComponentModel.DataAnnotations;
+using System.Threading;
 using Motor.Extensions.Hosting.RabbitMQ.Options;
 using RabbitMQ.Client;
 
 namespace Motor.Extensions.Hosting.RabbitMQ
 {
-    public interface IRabbitMQConnectionFactory
+    public interface IRabbitMQConnectionFactory<T> : IDisposable
```

Comment:
I don't quite see why the connection factory needs to be parameterized. Can you elaborate what is the key part of this PR that make this necessary?

Reply:
This is necessary because this way the factory is registered for the same message type within the DI container and therefore can be registered as singleton without interfering with other consumers but is resolved to the same instance in the actual consumer and the monitor.

The connection sharing to avoid appearing of the monitor as another consumer requires this change.